main/src/main/scala/sbt/coursierint/CoursierRepositoriesTasks.scala

```diff
@@ -57,16 +57,7 @@ object CoursierRepositoriesTasks {
   private final val keepPreloaded = false // coursierKeepPreloaded.value
 
   def coursierResolversTask: Def.Initialize[sbt.Task[Seq[Resolver]]] = Def.task {
-    val bootResOpt = bootResolvers.value
-    val overrideFlag = overrideBuildResolvers.value
-    val result0 = bootResOpt.filter(_ => overrideFlag) match {
-      case Some(r) => r
-      case None =>
-        val extRes = externalResolvers.value
-        val isSbtPlugin = sbtPlugin.value
-        if (isSbtPlugin) sbtResolvers.value ++ extRes
-        else extRes
-    }
+    val result0 = fullResolvers.value.filterNot(_ == projectResolver.value)
```

Comment:
The filter does not mandatory as https://github.com/coursier/sbt-coursier/blob/fb19697c573489d1ddbd6c3afd9d8609a70c4fec/modules/lm-coursier/src/main/scala/lmcoursier/internal/Resolvers.scala#L114-L115 effectively ignores `projectResolver`, but I thought it was better not to change the behavior of `csrResolvers`, particularly if this is backported to 1.6.x.